fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/ResourceFactoryImpl.java

```diff
@@ -57,64 +60,75 @@
+    public FedoraResource getResource(final Transaction transaction, final FedoraID fedoraID)
             throws PathNotFoundException {
-        return instantiateResource(transaction, identifier, version);
+        final FedoraResource resource = instantiateResource(transaction, fedoraID);
+        if (fedoraID.isTimemap()) {
+            return resource.getTimeMap();
+        }
+        return resource;
     }
 
     @Override
-    public <T extends FedoraResource> T getResource(final String identifier, final Class<T> clazz)
-            throws PathNotFoundException {
+    public <T extends FedoraResource> T getResource(final FedoraID identifier,
+                                                    final Class<T> clazz) throws PathNotFoundException {
         return clazz.cast(getResource(null, identifier));
     }
 
     @Override
-    public <T extends FedoraResource> T getResource(final Transaction transaction, final String identifier,
+    public <T extends FedoraResource> T getResource(final Transaction transaction, final FedoraID identifier,
             final Class<T> clazz) throws PathNotFoundException {
         return clazz.cast(getResource(transaction, identifier));
     }
 
     @Override
-    public <T extends FedoraResource> T getResource(final Transaction transaction, final String identifier,
-            final Instant version, final Class<T> clazz) throws PathNotFoundException {
-        return clazz.cast(getResource(transaction, identifier, version));
-    }
-
-    @Override
-    public boolean doesResourceExist(final Transaction transaction, final String fedoraId, final Instant version) {
-        // TODO: Check the index first.
-
-        final PersistentStorageSession psSession = getSession(transaction);
-
-        try {
-            psSession.getHeaders(fedoraId, version);
+    public boolean doesResourceExist(final Transaction transaction, final FedoraID fedoraId) {
+        if (fedoraId.isRepositoryRoot()) {
+            // Root always exists.
             return true;
-        } catch (final PersistentItemNotFoundException e) {
-            // Object doesn't exist.
-            return false;
-        } catch (final PersistentStorageException e) {
-            // Other error, pass along.
-            throw new RepositoryRuntimeException(e);
-        } finally {
-            if (transaction == null) {
-                // Commit session (if read-only) so it doesn't hang around.
-                try {
-                    psSession.commit();
-                } catch (final PersistentStorageException e) {
-                    LOGGER.error("Error committing session, message: {}", e.getMessage());
+        }
+        final FedoraID actualId;
+        if (fedoraId.isDescription()) {
+            // Descriptions are actually part of the binary, so use its ID.
+            actualId = FedoraID.create(fedoraId.getResourceId().replace("/" + FCR_METADATA, ""));
```

Comment:
So the issue is that the `fcr:metadata` is not needed for the resourceExists because we don't track the description in the containment index, which is why we strip it off. But [here](https://github.com/fcrepo4/fcrepo4/pull/1652/files#diff-6bd6c97381bbf8b748236dcffede84baR173) we call `getResourceId()` but for descriptions we need that `fcr:metadata`. Perhaps I'll change that spot and have it return `getFullId()` for descriptions.